### PR TITLE
add model and mujoco meshes

### PR DESCRIPTION
**Pull request by wuphilipp**
* add mujoco xml and meshes

* add simple interface and example script for using the mujoco model

### CODE REVIEW

File: opthex/hex.py [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+        """
+        raise NotImplementedError
+
+
+class MujocoHexRobot(HexRobot):
+
+    """The hexapod robot in mujoco."""
+
+    def __init__(self, load_path: str = "../models/hex.xml", viewer: bool = True):
+        """Initialize a hex robot in mujoco."""
```
Comment (brentyi) [inline opthex/hex.py]: missing input docstring

File: opthex/hex.py [hunk 2]
```diff
@@ -0,0 +1,122 @@
+import numpy as np
+import os
+import abc
+from typing import Dict
+from dataclasses import dataclass
+
+from mujoco_py import load_model_from_path, MjSim, MjViewer
```
Comment (brentyi) [inline opthex/hex.py]: can we commit to import sorting?

shameless plug: https://github.com/brentyi/dotfilesp/blob/master/common/.vimrc#L555-L563

File: opthex/hex.py [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+
+
+@dataclass(frozen=True)
+class HexState:
+    """Encompasses the current state of the robot."""
+
+    joints: Dict[str, float]
+    """Current joint state of the robot. A mapping between the joint name to its value."""
```
Comment (brentyi) [inline opthex/hex.py]: this looks longer than 88 characters (black's max line length)

File: opthex/hex.py [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+        """
+        sim_state = self.sim.get_state()
+        for name, value in joints.items():
+            joint_id = self.joint_name_id_map[name]
+            sim_state.qpos[joint_id] = value
+        self.sim.set_state(sim_state)
+        self.sim.forward()
+
+    def step(self) -> None:
+        """Steps the simulation forward by one step. Updates the visualizer if one is available."""
```
Comment (brentyi) [inline opthex/hex.py]: also looks longer than 88 characters

File: opthex/hex.py [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+
+        Parameters
+        ----------
+        command : Dict[str, float]
+            The commands to the robot.
+        """
+        ctrl = np.zeros((6,))
+        for name, value in command.items():
+            ctrl[self.joint_actuator_id_map[name]] = value
+        self.sim.data.ctrl[:] = ctrl
```
Comment (brentyi) [inline opthex/hex.py]: just wondering:
- is there a reason this needs to be in-place?
- would it make sense to modify `self.`sim.data.ctrl` directly, rather than creating a new array?

File: tests/test_hex.py [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+import hypothesis.extra.numpy as hnp
+from hypothesis import given
+from typing import Any
+from collections import OrderedDict
+
+from opthex.hex import HexState
+
+
+@st.composite
+def hex_state_stragety(draw: Any) -> HexState:
```
Comment (brentyi) [inline tests/test_hex.py]: typo

File: opthex/hex.py (hunk 2 above)
Comment (wuphilipp) [reply opthex/hex.py]: sounds good

File: tests/test_hex.py [hunk 7]
```diff
@@ -0,0 +1,40 @@
+import sys
+from unittest.mock import Mock
+
+sys.modules["mujoco_py"] = Mock()  # mock mujoco
```
Comment (wuphilipp) [inline tests/test_hex.py]: This is going to remain unsorted. Mujoco cant be imported anywhere on in the github workflow because otherwise, it will complain because there is no mujoco key. Not sure if there is a better workaround.

Comment (brentyi) [reply tests/test_hex.py]: that makes sense!

I added an explicit sort ignore line for this:
https://github.com/brentyi/opthex/pull/3/commits/9673ebdfb0dd5a6fb9fbb5c63564d67aee1c2e56

File: tests/test_hex.py [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+    acceleration = draw(hnp.arrays(dtype=np.float32, shape=3))
+    return HexState(
+        qpos=OrderedDict(zip(joint_names, joint_pos)),
+        qvel=OrderedDict(zip(joint_names, joint_vel)),
+        acceleration=acceleration,
+    )
+
+
+@given(hex_state=hex_state_strategy())
+def test_hex_state_joint_state(hex_state: HexState) -> None:
```
Comment (brentyi) [inline tests/test_hex.py:31]: just wondering: is there an advantage of using hypothesis instead of native pytest fixtures?

**Review by brentyi — APPROVED** [review]

File: tests/test_hex.py (hunk 8 above)
Comment (wuphilipp) [reply tests/test_hex.py:31]: Hypothesis will generate a variety of different test cases (100 by default) and will search for edge cases automatically so you won't have to think about it. It also has some other fancy features like giving you the simplest failure case it can find. In this case, it's probably overkill but could be useful later when we have more tools that consume the state.
